android/beagle/src/main/java/br/com/zup/beagle/android/components/page/PageView.kt

```diff
@@ -58,24 +93,35 @@ data class PageView(
         val container = viewFactory.makeBeagleFlexView(rootView.getContext(), style).apply {
             addView(viewPager, style)
         }
-
         pageIndicator?.let {
             val pageIndicatorView = viewRendererFactory.make(it).build(rootView)
             setupPageIndicator(children.size, viewPager, pageIndicator)
             container.addView(pageIndicatorView)
         }
+        viewPager.configPageChangeListener(
+            {
+                pageIndicator?.onItemUpdated(it)
+            },
+            {
+                viewPager.executeActions(rootView, it)
+            }
+        )
 
+        viewPager.observerCurrentPage(rootView)
         return container
     }
 
-    private fun setupPageIndicator(
-        pages: Int,
-        viewPager: BeaglePageView,
-        pageIndicator: PageIndicatorComponent?
+    private fun BeaglePageView.executeActions(rootView: RootView, position: Int){
```

Comment:
I think you don't need to create an extension function to do this only a private method resolves this case.


What do you think about rename this method to reference the event:
```suggestion
    private fun BeaglePageView.executeOnChangeActions(rootView: RootView, position: Int){
```

Reply:
I'll change, and suggest accepted